backend/tracim_backend/lib/utils/authentication.py

```diff
@@ -20,16 +28,195 @@
+                app_config.IDP_LIST.append(data.common_identifier)
+
+        configurator.add_forbidden_view(self._idp_chooser)
+        configurator.add_route("acs", "/saml/acs", request_method="POST")
+        configurator.add_route("sso", "/saml/sso", request_method="GET")
+        configurator.add_route("metadata", "/saml/metadata", request_method="GET")
+        configurator.add_view(self._acs, route_name="acs")
+        configurator.add_view(self._sso, route_name="sso")
+        configurator.add_view(self._metadata, route_name="metadata")
+
+    def _load_settings(self, config: Configurator):
+        # TODO - M.L - 2023/09/05 - Change this to proper config loading
+        """Parse and validate SAML configuration.
+
+        Args:
+            config (pyramid.config.Configurator): The configuration of the
+                existing Pyramid application.
+        """
+
+        if 'PYRAMID_SAML_PATH' in os.environ:
+            if 'pyramid_saml' not in config.get_settings():
+                config.get_settings().update({
+                    'pyramid_saml': {}
+                })
+            config.get_settings().get('pyramid_saml').update({
+                'saml_path': os.environ.get('PYRAMID_SAML_PATH')
+            })
+
+        if 'pyramid_saml' not in config.get_settings():
+            msg = 'Missing \'pyramid_saml\' configuration in settings'
+            raise AssertionError(msg)
+        else:
+            settings = config.get_settings().get('pyramid_saml')
+
+        if 'saml_path' not in settings:
+            msg = 'Missing \'saml_path\' in settings'
+            raise AssertionError(msg)
+
+    def authenticated_userid(
+        self, request: TracimRequest
+    ) -> typing.Optional[typing.Union[str, int]]:
+        # FIXME - SGD - 2023-07-26 - Return Tracim user id when SAML data is present in the
+        #  session. By matching it with SAML ID attribute (in external_id,
+        #  see https://github.com/tracim/tracim/issues/6209)
+        saml_user_id = request.session.get("saml_user_id")
+        if saml_user_id is None:
+            return None
+
+        user_api = UserApi(None, request.dbsession, self.app_config)
+
+        user = user_api.saml_authenticate(
+            user=None,
+            user_id=saml_user_id,
+            name=request.session.get("saml_name"),
+            mail=request.session.get("saml_mail"),
+        )
+        self.remember(request, user.user_id)
+        request.set_user(user)
+        return user.user_id
+
+    def permits(self, request, context, permission):
+        # TODO - M.L - 2023/09/06 - Maybe check for session lifetime or validity
+        # if request.authenticated_userid is None \
+        #     and not request.path.startswith("/saml") \
+        #     and not request.path.startswith("/saml") \
+        #     and not request.path is "/ui/login":
+        #     return Denied("Nobody is allowed")
+        return Allowed("Allowed")
```

Comment:
permits just, uh permits at the time. The initial blocking pattern was too exclusive and blocked access to too many ressources. A proper check of the session lifetime or validity must be done here.